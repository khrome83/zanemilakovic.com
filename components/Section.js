// @ts-check --jsx

import PropTypes from 'prop-types';

const Section = ({ children, className, dark }) => (
  <section className={`section ${className} ${dark && 'dark'}`}>
    <div className="content">
      { children }
    </div>
    <style jsx>{`
      @import './styles/variables.css';

      .section {
        position: relative;
        padding: 0 var(--outer-padding-mobile);
        border-top: 1px solid transparent;

        @media (--tablet-viewport) {
          padding: 0 var(--outer-padding-tablet);
        }

        @media (--desktop-viewport) {
          padding: 0 var(--outer-padding-desktop);
        }
      }

      .dark {
        background-color: brown;

        & :global(*) {
          color: var(--page-font-color-light);
        }
      }

      .content {
        margin: 0 auto;
        max-width: var(--max-content-width);
      }
    `}</style>
  </section>
);

Section.defaultProps = {
  className: null,
  dark: false,
};

Section.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
  dark: PropTypes.bool,
};

const SectionFullBleed = ({ children, className }) => (
  <section className={`section ${className}`}>
    { children }
    <style jsx>{`
      @import './styles/variables.css';

      .section {
        position: relative;
        width: 100vw;
        max-width: var(--max-bleed-width);
      }

      .masthead {
        height: 100vw; /* 1:1 */

        @media (--tablet-viewport) {
          height: 75vw; /* 4:3 */
        }

        @media (--desktop-viewport) {
          height: 56.25vw; /* 16:9 */
        }
      }
    `}</style>
  </section>
);

SectionFullBleed.defaultProps = {
  className: null,
};

SectionFullBleed.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
};

const Overlay = ({ children }) => (
  <div className="overlay">
    { children }
    <style jsx>{`
      .overlay {
        position: absolute;
        width: 100%;
        top: 0;
        left: 0;
        right: 0;
        bottom: 0;
        background-color: rgba(0, 0, 0, 0.5);
      }
    `}</style>
  </div>
);

Overlay.propTypes = {
  children: PropTypes.node.isRequired,
};

export { Section as default, SectionFullBleed, Overlay };
