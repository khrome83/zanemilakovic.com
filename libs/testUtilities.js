// @ts-check --jsx
/* eslint-disable no-param-reassign */

const cleanSnapshots = (node) => {
  for (const prop in node) {
    if (prop === 'className') {
      const className = node[prop].replace(/jsx-[0-9]+[\s]?/g, '');
      node.className = className || '';
      if (!node.className) {
        delete node.className;
      }
    } else if (typeof node[prop] === 'object') {
      cleanSnapshots(node[prop]);
    }
  }
};

const lib = {
  cleanSnapshots,
};

export { lib as default, cleanSnapshots };
