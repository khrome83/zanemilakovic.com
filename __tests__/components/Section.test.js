import React from 'react';
import renderer from 'react-test-renderer';
import { cleanSnapshots } from '../../libs/testUtilities';
import Section, { SectionFullBleed, Overlay } from '../../components/Section';

test('Section renders correctly', () => {
  const component = renderer.create(<Section>Content Goes Here</Section>);
  const tree = component.toJSON();
  cleanSnapshots(tree);
  expect(tree).toMatchSnapshot();
});

test('SectionFullBleed renders correctly', () => {
  const component = renderer.create(<SectionFullBleed>Content Goes Here</SectionFullBleed>);
  const tree = component.toJSON();
  cleanSnapshots(tree);
  expect(tree).toMatchSnapshot();
});

test('Overlay renders correctly', () => {
  const component = renderer.create(<Overlay>Content Goes Here</Overlay>);
  const tree = component.toJSON();
  cleanSnapshots(tree);
  expect(tree).toMatchSnapshot();
});
