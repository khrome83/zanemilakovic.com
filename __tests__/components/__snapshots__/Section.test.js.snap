// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overlay renders correctly 1`] = `
<div
  className="overlay"
>
  Content Goes Here
</div>
`;

exports[`Section renders correctly 1`] = `
<section
  className="section null false"
>
  <div
    className="content"
  >
    Content Goes Here
  </div>
</section>
`;

exports[`SectionFullBleed renders correctly 1`] = `
<section
  className="section null"
>
  Content Goes Here
</section>
`;
